src/solana/SolanaEngine.ts

```diff
@@ -448,6 +557,12 @@ export class SolanaEngine extends CurrencyEngine<
     })
     edgeTransaction.signedTx = solTx.serialize().toString('base64')
     this.warn(`signTx\n${cleanTxLogs(edgeTransaction)}`)
+
+    // Remove all false values from addressCache
+    this.addressCache = new Map(
+      Array.from(this.addressCache.entries()).filter(([_, value]) => value)
+    )
```

Comment:
Optional: This code is creating 2 extra arrays and 1 extra map. Doing it in a loop won't create any extra data structures, so it should be a lot simpler & faster:

```ts
for (const [key, value] of this.addressCache.entries()) {
  if (!value) this.addressCache.delete(key)
}
```

Reply:
Oh, that makes sense.